AddressBook.java

```diff
@@ -0,0 +1,1317 @@
+     * Note that the type of the variable below can also be declared as List<String[]>, as follows:
+     *    private static final List<String[]> ALL_PERSONS = new ArrayList<>()
+     * That is because List is an interface implemented by the ArrayList class.
+     * In this code we use ArrayList instead because we wanted to to stay away from advanced concepts
+     * such as interface inheritance.
+     * ====================================================================================================
+     */
+    /**
+     * List of all persons in the address book.
+     */
+    public static enum personProperty {NAME, EMAIL, PHONE};
+    private static final ArrayList<HashMap<personProperty, String>> ALL_PERSONS = new ArrayList<HashMap<personProperty, String>>();
+  
+    class hashCompare implements Comparator<HashMap<personProperty, String>>{
+    	public int compare(HashMap person1, HashMap person2){
+    		String two=(String) person2.get(personProperty.NAME);
+    		String one=(String) person1.get(personProperty.NAME);
+    		return one.compareTo(two);
+    	}
+    	public boolean equals(Object obj){
+    	   return this==obj;
+    	}
+    	}
+
+    /**
+     * Stores the most recent list of persons shown to the user as a result of a user command.
+     * This is a subset of the full list. Deleting persons in the pull list does not delete
+     * those persons from this list.
+     */
+    private static ArrayList<HashMap<personProperty, String>> latestPersonListingView = getAllPersonsInAddressBook(); // initial view is of all
+
+    /**
+     * The path to the file used for storing person data.
+     */
+    private static String storageFilePath;
+
+    /*
+     * ==============NOTE TO STUDENTS======================================
+     * Notice how this method solves the whole problem at a very high level.
+     * We can understand the high-level logic of the program by reading this
+     * method alone.
+     * ====================================================================
+     */
+    public static void main(String[] args) {
+        showWelcomeMessage();
+        if (args.length >= 2) {
+            showToUser(MESSAGE_INVALID_PROGRAM_ARGS);
+            exitProgram();
+        }
+
+        if (args.length == 1) {
+        	setupGivenFileForStorage(args[0]);        	
+        	/*
+            String filePath = args[0];
+			if (!isValidFilePath(filePath)) {
+			    showToUser(String.format(MESSAGE_INVALID_FILE, filePath));
+			    exitProgram();
+			}
+			
+			storageFilePath = filePath;
+			createFileIfMissing(filePath);
+			*/
+        }
+
+        if(args.length == 0) {
+        	setupDefaultFileForStorage();
+            /*showToUser(MESSAGE_USING_DEFAULT_FILE);
+			storageFilePath = DEFAULT_STORAGE_FILEPATH;
+			createFileIfMissing(storageFilePath);*/
+        	
+        }
+        
+        //loadDataFromStorage(); not sure how to do this
```

Comment:
Comments should be about the code only. It should still make sense to any future developer reading the comment. If you want to tell something to the reviewer/tutor (but not future developers of the code), use a GitHub comment instead (like this comment I've added).